### PR TITLE
[repo] docs: Revamp getting-started docs

### PR DESCRIPTION
**Pull request by singhk97**
## Linked issues

closes: #589 (issue number)

## Details

Update the getting started docs:

1. Aligned JS and .NET docs
2. Rewrite most sections to be simple and have a consistent format

#### Change details

> Describe your changes, with screenshots and code snippets as appropriate

### CODE REVIEW

**Review by aacebo — APPROVED** [review]

File: getting-started/01.TEMPLATES.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 1. [Prompts](./00.PROMPTS.md)
-2. [**Templates**](./01.TEMPLATES.md)
-3. [Actions](./02.ACTIONS.md)
-4. [Plans](./03.PLANS.md)
-5. [Chaining](./04.CHAINING.md)
-6. [Storage](./05.STORAGE.md)
-7. [Turns](./06.TURNS.md)
-
-A simple and powerful way to
+2. [**Prompt Template**](./01.TEMPLATES.md)
```
Comment (corinagum) [inline getting-started/01.TEMPLATES.md:4]: Did you want to rename this file to prompt templates? (optional)

File: getting-started/05.TURNS.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
+# Turns
```
Comment (corinagum) [inline getting-started/05.TURNS.md:1]: Was this one going to be TurnState?

File: getting-started/dotnet/00.MIGRATION.md [hunk 3]
```diff
@@ -1,20 +1,14 @@
-# Migrating from Botbuilder SDK to Teams AI.
+# Migrating from Bot Framework (BF)  SDK to Teams AI
```
Comment (corinagum) [inline getting-started/dotnet/00.MIGRATION.md:1]: Would you mind adding a TOC for the `dotnet` and `js` directories, same format as `getting-started`? Thank you~

File: getting-started/04.CHAIN.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+- createList name="<list name>"
+- addItem list="<list name>" item="<text>"
+
+User: Create a list called "grocery shopping" and add "bananas" to it. 
+```
+> See [Planner](./02.PLANNER.md) to learn about the planner prompt.
+
+And that you have orchestrated the `AI` module to use an LLM planner like `OpenAIPlanner`.
+
+Calling chain on this prompt will first use the `OpenAIPlanner` to generate a plan, that may look like:
```
Comment (corinagum) [inline getting-started/04.CHAIN.md:33]:
```suggestion
Calling chain on this prompt will first use the `OpenAIPlanner` to generate a plan that may look like:
```

File: getting-started/03.ACTIONS.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+The user can register custom actions or override default actions in the system. Below is a list of default actions present:
+
+| Action   | Description  |
+|---|---|
+| `__UnkownAction___`  | Called anytime an unknown action is predicted by the planner.   |
+| `___FlaggedInput___`  | Called anytime an input is flagged by the moderator.   | 
+| `___FlaggedOutput___`  | Called anytime an output is flagged by the moderator.   | 
+| `___RateLimited___`  | Called anytime the planner is rate limited.   | 
+| `___PlanReady___`  | Called after the plan has been predicted by the planner and it has passed moderation.   | 
+| `___DO___`  | Called to DO an action.   |
```
Comment (corinagum) [inline getting-started/03.ACTIONS.md:59]: I would leave a note that changing these will change ALL `DO` and `SAY` actions (as in, it's global as opposed to one-time).

**Review by corinagum — COMMENTED** [review]
Were changes/notes regarding updating the RAG sample txt files going to be in a different PR? Or still need to discuss/consider?

**Review by corinagum — DISMISSED** [review]
LGTM, please see questions/nits before merging ❤️

File: getting-started/05.TURNS.md (hunk 2 above)
Comment (singhk97) [reply getting-started/05.TURNS.md:1]: this one does cover the basics of the turn state which i think is suffices for now.

File: getting-started/03.ACTIONS.md (hunk 5 above)
Comment (singhk97) [reply getting-started/03.ACTIONS.md:59]: done

File: getting-started/dotnet/00.MIGRATION.md (hunk 3 above)
Comment (singhk97) [reply getting-started/dotnet/00.MIGRATION.md:1]: done

File: getting-started/01.TEMPLATES.md (hunk 1 above)
Comment (singhk97) [reply getting-started/01.TEMPLATES.md:4]: done

**Comment by singhk97** [comment]
The Teams Chef bot RAG index will be updated in a separate PR. The functionality of the bot won't be affect since the changes in the PR pretty much organizes the information and doesn't remove any crucial context.

**Review by corinagum — APPROVED** [review]

**Review by aacebo — APPROVED** [review]

**Pull request merged**